src/alternatives.js

```diff
@@ -0,0 +1,144 @@
+import has from 'lodash/has';
+import inherits from './util/inherits';
+import MixedSchema from './mixed';
+import { alternatives as locale } from './locale.js';
+import isSchema from './util/isSchema';
+import typeOf from 'type-name';
+import { SynchronousPromise } from 'synchronous-promise';
+import createValidation from './util/createValidation';
+import runValidations from './util/runValidations';
+
+let promise = sync => sync ? SynchronousPromise : Promise;
+const invalidMarker = {};
+
+
+export default function AlternativesSchema() {
+    if (!(this instanceof AlternativesSchema))
+        return new AlternativesSchema()
+
+    MixedSchema.call(this, { type: 'alternatives' })
+}
+const defaultShouldUseForCasting = () => true;
+
+inherits(AlternativesSchema, MixedSchema, {
+    _typeCheck(v){
+        if(this._oneOfTypeOptions){
+            const {
+                schemas
+            } = this._oneOfTypeOptions;
+            return !!schemas.find(it => it.isType(v));
+        }
+        return false
+    },
+
+    _validate(_value, options = {}) {
```

Comment:
This is still reimplementing to much logic, in fact you probably don't need to implement this at all now that `oneOfType` is implemented in terms of a normal validation. just letting the base mixed.prototype._validate get called will work.

Reply:
So I create the validation in a temp variable in next, how can I make sure the validation runs without implementing _validate ? Should I assign it directly to a func in next?

        var next = this.clone();
        next._oneOfType = createValidation({....

Reply:
you don't need to use createValidation at all, `next.test()` should be fine

Reply:
Ah, that cleans up 👍